src/Symfony/Component/HttpClient/Tests/NativeHttpClientTest.php

```diff
@@ -21,6 +21,28 @@ protected function getHttpClient(string $testCase): HttpClientInterface
         return new NativeHttpClient();
     }
 
+    public function testItCanBeNonBlockingStream()
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+        $response = $client->request('GET', 'http://localhost:8057');
+        $stream = $response->toStream();
+
+        $this->assertTrue(stream_get_meta_data($stream)['blocked']);
+        $this->assertTrue(stream_set_blocking($stream, 0));
+
+        // Help wanted. I've no idea why this test does not pass.
+        // $this->assertFalse(stream_get_meta_data($stream)['blocked']);
```

Comment:
I've no idea where data from `stream_get_meta_data` comes.

It looks like default values are always returned:
https://github.com/php/php-src/blob/9099dbd9614b37b48818ac24e2020b0d756b7e1e/ext/standard/streamsfuncs.c#L511-L515

Reply:
Correct, we cannot define these when using stream wrappers.